internal/impl/openai/image_processor.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openai
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"slices"
+
+	oai "github.com/Azure/azure-sdk-for-go/sdk/ai/azopenai"
+	"github.com/redpanda-data/benthos/v4/public/bloblang"
+	"github.com/redpanda-data/benthos/v4/public/service"
+)
+
+const (
+	oipFieldPrompt  = "prompt"
+	oipFieldQuality = "quality"
+	oipFieldSize    = "size"
+	oipFieldStyle   = "style"
+)
+
+func init() {
+	err := service.RegisterProcessor(
+		"openai_image_generation",
+		imageProcessorConfig(),
+		makeImageProcessor,
+	)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func imageProcessorConfig() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Categories("AI").
+		Summary("Processor that uses the OpenAI API to generate an image.").
```

Comment:
Generates an image from a text description and other attributes, using OpenAI API.